nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -428,37 +442,38 @@ protected String getCommentContainerPath(CoreSession session, String commentedDo
         });
     }
 
+    // former top level document method
     protected DocumentRef getAncestorRef(CoreSession session, DocumentModel documentModel) {
-        DocumentModel ancestorComment = getThreadForComment(session, documentModel);
-        return ancestorComment.getRef();
+        if (!documentModel.hasSchema(COMMENT_SCHEMA)) {
+            return documentModel.getRef();
+        }
+        // thread is the higher comment in a succession of replies
+        DocumentModel thread = getThreadForComment(session, documentModel);
+        return new IdRef((String) thread.getPropertyValue(COMMENT_PARENT_ID_PROPERTY));
     }
 
     protected DocumentModel getThreadForComment(CoreSession s, DocumentModel comment) throws CommentSecurityException {
         NuxeoPrincipal principal = s.getPrincipal();
         return CoreInstance.doPrivileged(s, session -> {
-            // Fetch the document in a case where it is not associated to an open session
-            DocumentModel documentModel = session.getDocument(comment.getRef());
-            while (documentModel.hasSchema(COMMENT_SCHEMA) || HIDDEN_FOLDER_TYPE.equals(documentModel.getType())) {
-                documentModel = session.getDocument(
-                        new IdRef((String) documentModel.getPropertyValue(COMMENT_PARENT_ID)));
+            DocumentModel thread = comment;
+            DocumentModel parent = s.getDocument(
+                    new IdRef((String) thread.getPropertyValue(COMMENT_PARENT_ID_PROPERTY)));
+            if (parent.hasSchema(COMMENT_SCHEMA)) {
+                thread = getThreadForComment(parent);
```

Comment:
Is there a reason to remove this version of the thread comment calculation and go back to the recursion once. As discussed initially the idea was to avoid using the recursive algo

Reply:
Yes, by reading it again, I wasn't able to say easily if the behavior has changed or not. After a moment of analysis, I though behavior has changed and as the class is deprecated I rollbacked.